app/javascript/reducers/portfolios.js

```diff
@@ -32,10 +34,40 @@ function fetchPortfolios(state, action) {
   }
 }
 
+function viewBuildingDetails(state, action) {
```

Comment:
Same comment from before about renaming Redux operations to something more action-like, for example `setActiveBuilding`